server/src-lib/Hasura/GraphQL/RemoteServer.hs

```diff
@@ -99,35 +99,48 @@ fetchRemoteSchema manager name def@(RemoteSchemaInfo url headerConf _ timeout) =
       Left _  -> J.object ["raw_body" J..= bsToTxt (BL.toStrict bs)]
 
 mergeSchemas
-  :: (MonadError QErr m)
+  :: (QErrM m)
   => RemoteSchemaMap
+  -> RemoteSchemasWithRole
   -> GS.GCtxMap
-  -- the merged GCtxMap and the default GCtx without roles
-  -> m (GS.GCtxMap, GS.GCtx)
-mergeSchemas rmSchemaMap gCtxMap = do
-  def <- mkDefaultRemoteGCtx remoteSchemas
-  merged <- mergeRemoteSchema gCtxMap def
-  return (merged, def)
+  -> m GS.GCtxMap
+mergeSchemas rmSchemaMap rmSchemaMapWithRole initGCtxMap = do
+  merged <- mergeRoleRemoteSchemas initGCtxMap rmSchemaMapWithRole
+  remotes <- combineRemoteGCtx remoteSchemas
+  finalMerged <- addRemoteSchemaToAdminRole merged remotes
+  return finalMerged
```

Comment:
`addRemoteSchemaToAdminRole merged remotes` in last line will work fine.

```haskell
= do
  b <- f a
  return b
```
can be rewritten as
```haskell
= f a
```